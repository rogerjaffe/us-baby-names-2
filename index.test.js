const byName = require('./yearsByName.json')
const byYear = require('./namesByYear.json')
// const byName = require('./by-name')
// const byYear = require('./by-year')
const kindOf = require('kind-of')

describe('byName', () => {
  describe('Mary', () => {
    const mary = byName['Mary']
    const firstRecord = mary[0]
    test('The first ten records match the previous snapshot', () => {
      const firstTenRecords = mary.slice(0, 10)
      expect(firstTenRecords.length).toEqual(10)
      expect(firstTenRecords).toMatchSnapshot()
    })
    test("The first record's count property is a number >= 5", () => {
      expect(firstRecord).toHaveProperty('count')
      expect(kindOf(firstRecord.count)).toEqual('number')
      expect(firstRecord.count).toBeGreaterThanOrEqual(5)
    })
    test("The first record's year property is a number >= 1880", () => {
      expect(firstRecord).toHaveProperty('year')
      expect(kindOf(firstRecord.year)).toEqual('number')
      expect(firstRecord.count).toBeGreaterThanOrEqual(1880)
    })
    test("The first record's sex property is a string", () => {
      expect(firstRecord).toHaveProperty('sex')
      expect(kindOf(firstRecord.sex)).toEqual('string')
    })
  })
})

describe('byYear', () => {
  describe('2021', () => {
    const twentySixteen = byYear[2021]
    const firstRecord = twentySixteen[0]
    test('The first ten records match the previous snapshot', () => {
      const firstTenRecordsOf2016 = twentySixteen.slice(0, 10)
      expect(firstTenRecordsOf2016.length).toEqual(10)
      expect(firstTenRecordsOf2016).toMatchSnapshot()
    })
    test("The first record's count property is a number >= 5", () => {
      expect(firstRecord).toHaveProperty('count')
      expect(kindOf(firstRecord.count)).toEqual('number')
      expect(firstRecord.count).toBeGreaterThanOrEqual(5)
    })
    test("The first record's name property is a string", () => {
      expect(firstRecord).toHaveProperty('name')
      expect(kindOf(firstRecord.name)).toEqual('string')
    })
    test("The first record's sex property is a string", () => {
      expect(firstRecord).toHaveProperty('sex')
      expect(kindOf(firstRecord.sex)).toEqual('string')
    })
  })
})
