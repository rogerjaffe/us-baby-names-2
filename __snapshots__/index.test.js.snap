// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`byName Mary The first ten records match the previous snapshot 1`] = `
[
  {
    "count": 7065,
    "sex": "F",
    "year": 1880,
  },
  {
    "count": 27,
    "sex": "M",
    "year": 1880,
  },
  {
    "count": 6919,
    "sex": "F",
    "year": 1881,
  },
  {
    "count": 29,
    "sex": "M",
    "year": 1881,
  },
  {
    "count": 8148,
    "sex": "F",
    "year": 1882,
  },
  {
    "count": 30,
    "sex": "M",
    "year": 1882,
  },
  {
    "count": 8012,
    "sex": "F",
    "year": 1883,
  },
  {
    "count": 32,
    "sex": "M",
    "year": 1883,
  },
  {
    "count": 9217,
    "sex": "F",
    "year": 1884,
  },
  {
    "count": 36,
    "sex": "M",
    "year": 1884,
  },
]
`;

exports[`byYear 2021 The first ten records match the previous snapshot 1`] = `
[
  {
    "count": 17728,
    "name": "Olivia",
    "sex": "F",
  },
  {
    "count": 15433,
    "name": "Emma",
    "sex": "F",
  },
  {
    "count": 13285,
    "name": "Charlotte",
    "sex": "F",
  },
  {
    "count": 12952,
    "name": "Amelia",
    "sex": "F",
  },
  {
    "count": 12759,
    "name": "Ava",
    "sex": "F",
  },
  {
    "count": 12496,
    "name": "Sophia",
    "sex": "F",
  },
  {
    "count": 11201,
    "name": "Isabella",
    "sex": "F",
  },
  {
    "count": 11096,
    "name": "Mia",
    "sex": "F",
  },
  {
    "count": 9434,
    "name": "Evelyn",
    "sex": "F",
  },
  {
    "count": 8388,
    "name": "Harper",
    "sex": "F",
  },
]
`;
